### DISCUSSION TITLE
rqlite_client library crate for Rust

### DISCUSSION THREAD

**Discussion by kolbma** [rqlite/rqlite#1407] [Show and tell]
Hey,

I've published the [__rqlite_client__ crate](https://crates.io/crates/rqlite_client) for using _rqlite_ backends via _API_ from [Rust](https://www.rust-lang.org/).  

There is the possibility to create type safe queries and retrieve type safe results. It has an optional implementation for database scheme migration and rollback. Per default the HTTP(S) requests are handled by a provided implementation based on crate [ureq](https://crates.io/crates/ureq). So simple `sync` requests.  
It provides also some logging for `log` or `tracing` crate.

Please have a look, try it out, give some suggestions for improvement!

Crates.io: <https://crates.io/crates/rqlite_client/>  
Docs.rs: <https://docs.rs/rqlite_client/latest/>
Github: <https://github.com/kolbma/rs_rqlite_client/>

**Comment by otoolep** [comment]
Thanks very much for your contribution. I'm not a Rust programmer, so do not have a client library for rqlite -- so this is great to see.

Would you be interested in moving this repo to the rqlite organization? You would have full commit rights to that repo (so you could maintain the repo as you see fit), but it would live under the rqlite org. That way people would find it much easier.

**Reply by kolbma** [reply]
Nice offer to be part of rqlite-org. Thanks, too.
I've to think about and would like to stabilize the API a little bit before a move.  
So to say, when I'll start to use version >0.0, I think it could be the moment.  
At the moment I'm not thinking about any API stability and compatibility. What makes sense and is a gain, will be changed.  
I'll come back to this.

**Reply by otoolep** [reply]
Sounds good.

**Comment by otoolep** [comment]
I have created a new page on rqlite.io, which lists all known client libraries. Added yours to the list.

https://rqlite.io/docs/api/client-libraries/
